src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/PropertyDescriptor.cs

```diff
@@ -165,10 +170,11 @@ public virtual void AddValueChanged(object component, EventHandler handler)
             ArgumentNullException.ThrowIfNull(component);
             ArgumentNullException.ThrowIfNull(handler);
 
-            _valueChangedHandlers ??= new Dictionary<object, EventHandler?>();
-
-            EventHandler? h = _valueChangedHandlers.GetValueOrDefault(component, defaultValue: null);
-            _valueChangedHandlers[component] = (EventHandler?)Delegate.Combine(h, handler);
+            lock (SyncObject)
+            {
+                _valueChangedHandlers ??= new ConcurrentDictionary<object, EventHandler?>();
```

Comment:
Am I reading this correctly that there's one of these dictionaries possible per property? Are many writes expected? The default concurrency level for ConcurrentDictionary is Environment.ProcessorCount, meaning each of these CDs is going to have that many locks created for it under the covers at a minimum. If the thread-safety here is just precautionary and a lot of concurrency isn't expected, consider using the constructor that allows you to specify a concurrency level and set it to 1.

Reply:
Thanks. Changing to 1